src/QsFmt/App.Tests/Tests.fs

```diff
@@ -148,173 +176,126 @@ let ``Shows file not found error`` path =
-    let input =
-        "namespace StandardIn { function Bar() : Int { return 0; } }
-"
-
-    let output =
-        "namespace Example1 {
-    function Bar() : Int {
-        return 0;
-    }
-}
-namespace StandardIn {
-    function Bar() : Int {
-        return 0;
-    }
-}
-namespace SubExample1 {
-    function Bar() : Int {
-        return 0;
-    }
-}
-namespace SubExample2 {
-    function Bar() : Int {
-        return 0;
-    }
-}
-namespace SubExample3 {
-    function Bar() : Int {
-        return 0;
-    }
-}
-namespace NestedExample1 {
-    function Bar() : Int {
-        return 0;
-    }
-}
-namespace NestedExample2 {
-    function Bar() : Int {
-        return 0;
-    }
-}
-"
-
-    Assert.Equal(
+    let files =
+        [
+            Example1
+            SubExample1
+            SubExample2
+            SubExample3
+            NestedExample1
+            NestedExample2
+        ]
+
+    [|
+        "update"
+        "-r"
+        Example1.Path
+        "-"
+        "Examples\\SubExamples1"
+        "Examples\\SubExamples2"
+    |]
+    |> runWithFiles
+        true
+        files
+        StandardInputTest.Original
         {
             Code = 0
-            Out = output |> standardizeNewLines
+            Out = StandardInputTest.Updated
             Error = ""
-        },
-        run [| "format"; "-r"; "Example1.qs"; "-"; "SubExamples1"; "SubExamples2" |] (input |> standardizeNewLines)
-    )
+        }
 
 [<Fact>]
 let ``Process correct files while erroring on incorrect`` () =
-    let input =
-        "namespace StandardIn { function Bar() : Int { return 0; } }
-"
+    let files = [ Example1; Example2 ]
 
-    let output =
-        "namespace Example1 {
-    function Bar() : Int {
-        return 0;
-    }
-}
-namespace StandardIn {
-    function Bar() : Int {
-        return 0;
-    }
-}
-namespace Example2 {
-    function Bar() : Int {
-        return 0;
-    }
-}
+    try
+        let result =
+            run [| "update"; Example1.Path; "-"; "Examples\\NotFound.qs"; Example2.Path |] StandardInputTest.Original
+
+        Assert.Equal(3, result.Code)
+        Assert.Equal(StandardInputTest.Updated, result.Out)
+        Assert.NotEmpty(result.Error)
+
+        for file in files do
+            let after = File.ReadAllText file.Path |> standardizeNewLines
+            Assert.Equal(file.Updated, after)
+    finally
+        for file in files do
+            File.WriteAllText(file.Path, file.Original)
+
+[<Fact>]
+let ``Backup flag`` () =
+    let files = [ Example1; SubExample3 ]
+
+    try
+        let result = run [| "update"; "-b"; "-"; Example1.Path; "Examples\\SubExamples2" |] StandardInputTest.Original
+
+        Assert.Equal(
+            {
+                Code = 0
+                Out = StandardInputTest.Updated
+                Error = ""
+            },
+            result
+        )
+
+        for file in files do
+            Assert.True(File.Exists(file.Path + "~"))
```

Comment:
This pattern of adding "~" to the path repeats enough that it might be worth adding that as a function, which will also have the advantage that if we update the backup file naming pattern later there will be one place to change. It's only in this one test so far, so it's not a big deal at the moment and it's up to you if you want to refactor that now or leave as-is for the time being.

Reply:
Hmm, it is hard to consolidate logic that is referenced so sparsely. Outside of this test, the tilde only comes up once in the Program file. I've consolidated the logic a bit in the test, but I'll leave the rest alone. Future use cases would better inform how that bridge should be crossed.